network/server/tchannelthrift/node/service.go

```diff
@@ -74,19 +75,27 @@ type service struct {
 	nowFn   clock.NowFn
 	metrics serviceMetrics
 	health  *rpc.NodeHealthResult_
+	idpool  *ts.Pool
 }
 
 // NewService creates a new node TChannel Thrift service
 func NewService(db storage.Database) rpc.TChanNode {
-	opts := db.Options()
-	iops := opts.InstrumentOptions()
-	scope := iops.MetricsScope().SubScope("service").Tagged(map[string]string{"serviceName": "node"})
+	iopts := db.Options().InstrumentOptions()
+
+	scope := iopts.MetricsScope().SubScope("service").Tagged(
+		map[string]string{"serviceName": "node"},
+	)
 
 	return &service{
 		db:      db,
-		nowFn:   opts.ClockOptions().NowFn(),
-		metrics: newServiceMetrics(scope, iops.MetricsSamplingRate()),
-		health:  &rpc.NodeHealthResult_{Ok: true, Status: "up", Bootstrapped: false},
+		nowFn:   db.Options().ClockOptions().NowFn(),
+		metrics: newServiceMetrics(scope, iopts.MetricsSamplingRate()),
+		health: &rpc.NodeHealthResult_{
+			Ok:           true,
+			Status:       "up",
+			Bootstrapped: false,
+		},
+		idpool: ts.NewPool(pool.NewObjectPoolOptions()),
```

Comment:
We should this pool become a part of the `storage.Options` so others can define the pool size, metrics, etc and it is accessible to the rest of storage.